### PR TITLE
Add custom userdata to websocket obj

### PR DESCRIPTION
**Pull request by ADiea**
Sometimes we need extra information(defined in the application) to be stored for every websocket for purpose of connection reusing. This defines the possibility to store a void* that is treated as application intends.

### CODE REVIEW

**Comment by slaff** [comment]
@ADiea Can you change the Websocket sample and add a demo of the custom data in there?

**Comment by ADiea** [comment]
Sure. Will also add documentation for member functions, forgot.

File: Sming/SmingCore/Network/WebSocket.h [hunk 1]
```diff
@@ -34,13 +34,17 @@ class WebSocket
...
+	void *getUserData();
+
 protected:
 	bool initialize(HttpRequest &request, HttpResponse &response);
 	bool is(HttpServerConnection* conn) { return connection == conn; };
 
 private:
 	HttpServerConnection* connection;
 	CommandExecutor* commandExecutor = nullptr;
+	void* m_pUserData = nullptr;
```
Comment (slaff) [inline Sming/SmingCore/Network/WebSocket.h]: Coding Style: Can you rename `m_pUserData` to `userData`?

**Comment by slaff** [comment]
@ADiea The PR now is only adding new documentation. If there is something more coming please add it. If not, rename the PR to mention the new value of this PR.

**Comment by ADiea** [comment]
@slaff put together a simplistic scenario, probably it does not illustrate too well. The idea is to store custom object on every connection. For example if there was a Vector of users like George. It depends on user what stores in the void*

File: Sming/SmingCore/Network/WebSocket.cpp [hunk 2]
```diff
@@ -75,3 +75,13 @@ void WebSocket::close()
 {
 	connection->close();
 }
+
+void WebSocket::setUserData(void* _userData)
```
Comment (slaff) [inline Sming/SmingCore/Network/WebSocket.cpp]: Keep the coding style consistent. Rename '_userData' to 'userData'. Then use `this->userData = userData`

File: Sming/SmingCore/Network/WebSocket.h [hunk 3]
```diff
@@ -22,25 +22,70 @@ class WebSocket
...
+	 */
 	void setTimeOut(uint16_t waitTimeOut) { if(connection) connection->setTimeOut(waitTimeOut); };
 
+	/** @brief  Test if two sockets are the same
+	 *  @param  other The other websocket
+	 *  @return true if the two webSockets are defining the same connection
+	 */
 	bool operator==(const WebSocket &other) const { return this->connection == other.connection;};
 
+	/** @brief  Store user dtaa pointer for this socket(connection)
```
Comment (slaff) [inline Sming/SmingCore/Network/WebSocket.h]: Typo: `user dtaa` should be `user data`

File: Sming/SmingCore/Network/WebSocket.h [hunk 4]
```diff
@@ -22,25 +22,70 @@ class WebSocket
...
+	 *  @param  _userData Pointer to user defined data
+	 */
+	void setUserData(void* _userData);
```
Comment (slaff) [inline Sming/SmingCore/Network/WebSocket.h]: _userData => userData

File: samples/HttpServer_WebSockets/app/application.cpp [hunk 5]
```diff
@@ -10,6 +10,74 @@
 HttpServer server;
 int totalActiveSockets = 0;
 
+//Simplified container modelling a user session
+class CUserData
```
Comment (anakod) [inline samples/HttpServer_WebSockets/app/application.cpp]: Please put classes to extarnal files. It's important to keep samples as simple as possible.

**Comment by slaff** [comment]
@ADiea Can you please check anakod's comment on this PR, make the recommended changes and ping me so that I can merge it?

**Comment by ADiea** [comment]
hopefully tonight sorry out of dev time

**Pull request merged**